app/domain/authentication/authn_oidc/authenticate_oidc_conjur_token/validate_and_decrypt_oidc_conjur_token.rb

```diff
@@ -12,45 +12,42 @@ module AuthenticateOidcConjurToken
       # - Validate input
       # - Validate JWT signing token
       # - Decrypt ID token
-      class ValidateAndDecryptOidcConjurToken
-        extend CommandClass::Include
-
-        command_class(
-          dependencies: {},
-          inputs: %i(request_body)
-        ) do
-
-          def call
-            oidc_conjur_token
-            # validate_signing
-            # decrypt_token
-          end
-
-          private
-
-          def oidc_conjur_token
-            ::Authentication::AuthnOidc::ConjurToken.new(
-              id_token_encrypted: id_token_encrypted,
-              user_name: user_name,
-              expiration_time: expiration_time
-            )
-          end
-
-          def decoded_body
-            @decoded_body ||= URI.decode_www_form(@request_body)
-          end
-
-          def id_token_encrypted
-            @id_token_encrypted ||= decoded_body.assoc('id_token_encrypted').last
-          end
-
-          def user_name
-            @user_name ||= decoded_body.assoc('user_name').last
-          end
-
-          def expiration_time
-            @expiration_time ||= decoded_body.assoc('expiration_time').last
-          end
+
+      ValidateAndDecryptOidcConjurToken = CommandClass.new(
+        dependencies: {},
```

Comment:
You may want to do this separately, since it's not strictly related to the purpose of this PR,
but we should never have a CommandClass with no dependencies.  Or (a side point) names with the word `And` or `Or` in them.

In this particular case, it looks like the logic here belongs as a secondary constructor on `ConjurToken`.  That is, the primary or canonical way to construct a ConjurToken is by giving it an `id_token_encrypted`, `user_name` and `expiration_time`.

This is alternative way to construct using the contents of a request body (which contains those 3 bits of information, but in a different format).  Thus the secondary constructor would pull out the 3 bits of information, then pass them to the primary ctor, as we do here.

So I'd recommend creating `self.from_request_body` on `ConjurToken`, moving this logic there, and then deleting this class and file.

Reply:
this looks right but this class isn't used in the current OIDC scope (and is actually blocked in `routes.rb`. i created an issue for this in GH so we'll get to it when we start working on this.